cli/src/commands/deploy.ts

```diff
@@ -220,10 +227,10 @@ export default class Deploy extends Command {
 
     this.startStage('register app on local');
     const env = 'default'; // hardcoded for now
-    let project = findProjectByDirectory(projects, projectDir);
+    let project = forceNewApp ? undefined : findProjectByDirectory(projects, projectDir);
 
     let application: Application;
-    if (!project || givenAppName) {
+    if (!forceNewApp && (!project || givenAppName)) {
```

Comment:
This logic is getting _very_ hairy.  In particular, `forceNewApp` only ever appears in the negative... which makes for interesting reading.

At a micro level, I *think* this conditional might be easier to read in its negated dual form:
```suggestion
    if (!(forceNewApp || (project && !givenAppName)() {
```